internal/services/streamanalytics/stream_analytics_stream_input_eventhub_resource.go

```diff
@@ -72,14 +72,14 @@ func resourceStreamAnalyticsStreamInputEventHub() *pluginsdk.Resource {
 
 			"shared_access_policy_key": {
```

Comment:
Hi @neil-yechenwei, thank you for the Fix, but I would suggest to include the following
```
ConflictsWith: []string{
				"authentication_mode",
			},
```
Same for `shared_access_policy_name` and for `authentication_mode` where the other two are the conflicts.
See /internal/services/appservice/windows_function_app_resource.go#L128 as a reference.

Reply:
Nope. I assume `shared_access_policy_name` and `shared_access_policy_key` are required when `authentication_mode` is `ConnectionString`. So we cannot add ConflictsWith for them.

![image](https://user-images.githubusercontent.com/19754191/197913020-c1265aef-9d54-4b01-a885-918fc6544cf7.png)